www/__tests__/class.test.tsx

```diff
@@ -0,0 +1,75 @@
+import "@testing-library/jest-dom";
+
+import { renderSPA, setupSPAServer } from "@/lib/test/spa";
+
+setupSPAServer();
+
+describe("Class page", () => {
+  it("loads and shows correct information", async () => {
+    const { getByRole, getByText } = await renderSPA(
+      "/project-alpha/alpha.core/Helper"
+    );
+
+    // Project heading
+    const heading = getByRole("heading", { name: "Project project-alpha" });
+    expect(heading.children[0]).toHaveAttribute("href", "/project-alpha");
+
+    // Variable / method links aren't happy yet, see https://github.com/zsol/py.wtf/issues/3
+    expect(getByRole("link", { name: "some_variable" })).toHaveAttribute(
+      "href",
+      "/project-alpha/alpha.core/Helper"
+    );
+
+    // Inner class link
+    expect(getByRole("link", { name: "Utils" })).toHaveAttribute(
+      "href",
+      "/project-alpha/alpha.core/Helper.Utils"
+    );
+
+    // Other module member links, hopefully in the sidebar
+    for (const sym of ["Helper", "core_main"]) {
+      expect(getByRole("link", { name: sym })).toHaveAttribute(
+        "href",
+        `/project-alpha/alpha.core/${sym}`
+      );
+    }
+
+    // Class declaration
+    getByText(
+      (_, element) =>
+        element?.nodeName === "PRE" &&
+        element?.textContent === "class alpha.core.Helper()"
+    );
```

Comment:
What do these do? They don't look like being tests (there are no asserts) and they don't look to be utils (not used anywhere in the test suite)... Please teach me 🙂

Reply:
The `get*` queries explode if there are no matches; so these are kinda sorta tests that verify such an element exists.

Reply:
... I guess one could instead do `expect(findByRole(...)).to.notBeNull()`, matter of taste IMHO.